crates/auth/src/providers/impls/near_wallet.rs

```diff
@@ -536,9 +536,29 @@ impl AuthProvider for NearWalletProvider {
     }
 
     fn is_configured(&self) -> bool {
+        // NEAR wallet is configured if RPC URL is set
         !self.config.rpc_url.is_empty()
     }
 
+    async fn is_configured_with_users(&self) -> eyre::Result<bool> {
+        // For NEAR wallet, "configured" means both technically configured AND has users
+        if !self.is_configured() {
+            return Ok(false);
+        }
+
+        // Check if any root keys exist for this provider (auth_method = "near_wallet")
+        use crate::storage::models::KeyType;
+        let keys = self.key_manager.list_keys(KeyType::Root).await?;
+        // Check if any keys have the near_wallet auth method
+        let has_near_keys = keys.iter().any(|(_, key)| {
+            key.auth_method
+                .as_ref()
+                .map(|m| m == "near_wallet" || m == "near")
+                .unwrap_or(false)
+        });
```

Comment:
I think listing keys might be consumpting resources, idealy to peek and return early if it has at least any key. But then `key_manager` needs to have something like `has_any_key(prefix: KeyType::Root)` method. Or, alternatively, each key manager can keep track of `user_count`.